ui/prometheus-plugin/src/plugins/prometheus-time-series-query/PrometheusTimeSeriesQueryEditor.tsx

```diff
@@ -41,9 +43,18 @@ export function PrometheusTimeSeriesQueryEditor(props: PrometheusTimeSeriesQuery
   const datasourceSelectLabelID = `prom-datasource-label-${selectedDatasource.name || 'default'}`;
 
   const { data: client } = useDatasourceClient<PrometheusClient>(selectedDatasource);
-  const promURL = client?.options.datasourceUrl;
   const { data: datasourceResource } = useDatasource(selectedDatasource);
 
+  const promURL =
+    (datasourceResource?.plugin.spec as PrometheusDatasourceSpec).directUrl ?? client?.options.datasourceUrl;
+  let completeConfig: CompleteConfiguration;
+  if (datasourceResource && (datasourceResource.plugin.spec.proxy as HTTPProxy)?.spec.headers) {
```

Comment:
when the HTTP proxy is defined in the datasource, then it's the Perses backend server that is modifying the request before forwarding it to Prometheus.

So actually it shouldn't be necessary to do that. Injection of header should be done through the Proxy for security matter

Reply:
Hi, thank u for the review! This PR actually comes from a production problem in our web. We embedded perses panels in our workbench with datasource using directUrl and proxy of our http headers as a plugin spec. The proxy works fine with prometheus api for query_range, but didn't work with codemirror-promql's autocompletion (no http headers appear in the request). As I understand from this [line of code](https://github.com/perses/perses/pull/2370/files#diff-88f464b7edf2c44c2db875af4196defd6fe5fa870981627e195d6d35dfb2cd6eL83), the http request initiating from codemirror only have url specified from config, and didn't support the use of headers before the release of [codemirror-promql v0.49.0](https://www.npmjs.com/package/@prometheus-io/codemirror-promql/v/0.49.0).
Perhaps there are also other problems with the using of embedded panels and configuration of datasource using directUrl and proxy. Besides this PR, right now I can't figure out why the proxy only partially works.

Reply:
ah yeah ok I missed the part it was for the embedded part.
sounds good to me then. Thanks for the details and explanation :)